src/memex/models/document.py

```diff
@@ -393,29 +395,35 @@ def merge_documents(session, documents, updated=None):
             m.document = master
             m.updated = updated
 
-        session.delete(doc)
-
     try:
         session.flush()
+        session.query(Annotation) \
+            .filter(Annotation.document_id.in_(duplicate_ids)) \
+            .update({Annotation.document_id: master.id}, synchronize_session='fetch')
```

Comment:
That is a great point! I've added a bunch of annotation records in the tests for merge documents, but seem to have missed that there isn't a test that checks that! Will add right now.